go/stats/kebab_case_converter.go

```diff
@@ -22,6 +22,32 @@ import (
 	"sync"
 )
 
+// GetSnakeName calls toSnakeName on the passed in string. It produces
+// a snake-cased name from the provided camel-cased name.
+// It memoizes the transformation and returns the stored result if available.
+func GetSnakeName(name string) string {
+	return toSnakeCase(name)
+}
+
+// toSnakeCase produces a monitoring compliant name from the original.
+// For systems (like Prometheus) that ask for snake-case names.
+// It converts CamelCase to camel_case, and CAMEL_CASE to camel_case.
+// For numbers, it converts 0.5 to v0_5.
+func toSnakeCase(name string) (hyphenated string) {
```

Comment:
I added the toSnakeCase modifications based on the toKebabCase stuff.  I'm not totally convinced it structurally makes sense to put it here (at minimum, I should rename the file if it otherwise makes sense to you). 

Also, it shares a memoizer with the toKebabCase, which _should_ be fine (one build should only use one or the other), but perhaps they should be separate in the code too? 

@demmer, thoughts?

Reply:
For safety and cleanliness I actually don't think they should share a memoizer.

Given that... I don't think there's actually _any_ substantive shared code any more, so rather than misleadingly sharing the file name, I'd say you should just create a `snake_case_converter.go` with the new code and a separate memoizer.